code/modules/recycling/disposal/pipe.dm

```diff
@@ -0,0 +1,345 @@
+			new floorturf.floor_tile(T)
+		floorturf.make_plating()
+
+	if(direction)		// direction is specified
+		if(isspaceturf(T)) // if ended in space, then range is unlimited
+			target = get_edge_target_turf(T, direction)
+		else						// otherwise limit to 10 tiles
+			target = get_ranged_target_turf(T, direction, 10)
+
+		eject_range = 10
+
+	else if(floorturf)
+		target = get_offset_target_turf(T, rand(5)-rand(5), rand(5)-rand(5))
+
+	playsound(src, 'sound/machines/hiss.ogg', 50, 0, 0)
+	for(var/A in H)
+		var/atom/movable/AM = A
+		AM.forceMove(get_turf(src))
+		AM.pipe_eject(direction)
+		if(target)
+			AM.throw_at(target, eject_range, 1)
+	H.vent_gas(T)
+	qdel(H)
+
+
+// pipe affected by explosion
+/obj/structure/disposalpipe/contents_explosion(severity, target)
+	var/obj/structure/disposalholder/H = locate() in src
+	if(H)
+		H.contents_explosion(severity, target)
+
+
+/obj/structure/disposalpipe/run_obj_armor(damage_amount, damage_type, damage_flag = 0, attack_dir)
+	if(damage_flag == "melee" && damage_amount < 10)
+		return 0
+	return ..()
+
+
+//attack by item
+//weldingtool: unfasten and convert to obj/disposalconstruct
+/obj/structure/disposalpipe/attackby(obj/item/I, mob/user, params)
+	add_fingerprint(user)
+	if(istype(I, /obj/item/weldingtool))
+		if(!can_be_deconstructed(user))
+			return
+
+		var/obj/item/weldingtool/W = I
+		if(W.remove_fuel(0, user))
+			playsound(src, I.usesound, 50, 1)
+			to_chat(user, "<span class='notice'>You start slicing [src]...</span>")
+			// check if anything changed over 2 seconds
+			if(do_after(user, 30*I.toolspeed, target = src))
+				if(!W.isOn())
+					return
+				deconstruct()
+				to_chat(user, "<span class='notice'>You slice [src].</span>")
+	else
+		return ..()
+
+//checks if something is blocking the deconstruction (e.g. trunk with a bin still linked to it)
+/obj/structure/disposalpipe/proc/can_be_deconstructed()
+	return TRUE
+
+// called when pipe is cut with welder
+/obj/structure/disposalpipe/deconstruct(disassembled = TRUE)
+	if(!(flags_1 & NODECONSTRUCT_1))
+		if(disassembled)
+			if(stored)
+				stored.forceMove(loc)
+				transfer_fingerprints_to(stored)
+				stored.setDir(dir)
+				stored = null
+		else
+			var/turf/T = get_turf(src)
+			for(var/D in GLOB.cardinals)
+				if(D & dpdir)
+					var/obj/structure/disposalpipe/broken/P = new(T)
+					P.setDir(D)
+	qdel(src)
+
+
+/obj/structure/disposalpipe/singularity_pull(S, current_size)
+	..()
+	if(current_size >= STAGE_FIVE)
+		deconstruct()
+
+
+// Straight pipe segment
+/obj/structure/disposalpipe/segment
+	icon_state = "pipe"
+	initialize_dirs = DISP_DIR_FLIP
+
+/obj/structure/disposalpipe/segment/Initialize()
+	// Hacks for old map pipes to work, remove after all maps are updated
+	if(icon_state == "pipe-c")
+		switch(dir)
+			if(NORTH)
+				dir = NORTHEAST
+			if(SOUTH)
+				dir = SOUTHWEST
+			if(EAST)
+				dir = SOUTHEAST
+			if(WEST)
+				dir = NORTHWEST
+
+	if(icon_state != "pipe")
+		icon_state = "pipe"
+	. = ..()
```

Comment:
. = ..() goes at the top of Initialize if possible

Reply:
**Not possible**, this exact part is changing vars that are used in Init later. Also it's a hack to support old map pipes that will be removed in my next PR.